### PR TITLE
fix last-input in ff, & correct behaviour

### PR DESCRIPTION
**Pull request by chee**

### CODE REVIEW

**Comment by chee** [comment]
bep

File: app/assets/javascripts/lib/utils/last_input_device.js [hunk 1]
```diff
@@ -23,13 +20,13 @@ define(function() {
...
+
+    if (userDevices[deviceType]) return;
 
     // announce the presence of each device as it's used for the first time
-    if (!listener) { return; }
-    event = document.createEvent("CustomEvent");
-    event.initCustomEvent(":device/" + deviceType);
-    listener.dispatchEvent(event);
+    userDevices[deviceType] = true;
+    listener.dispatchEvent(new CustomEvent(":device/" + deviceType));
```
Comment (jcreamer898) [inline app/assets/javascripts/lib/utils/last_input_device.js]: Doesn't look like `CustomEvent` is supported in IE8? https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent

File: app/assets/javascripts/lib/utils/last_input_device.js [hunk 2]
```diff
@@ -23,13 +20,13 @@ define(function() {
       document.documentElement.className += " last-input-" + deviceType;
     }
 
-    userDevices[deviceType] = true;
+    if (!listener) return;
+
+    if (userDevices[deviceType]) return;
```
Comment (jcreamer898) [inline app/assets/javascripts/lib/utils/last_input_device.js]: Could combine these 2 if statements... 

```js
if (!listener || userDevices[deviceType]) return;
```

File: app/assets/javascripts/lib/utils/last_input_device.js (hunk 1 above)
Comment (jcreamer898) [inline app/assets/javascripts/lib/utils/last_input_device.js]: Never mind, realizing this code only affects "modern" browsers.

**Comment by remybach** [comment]
:+1:

**Pull request merged**